### DISCUSSION TITLE
set geoip.dat file location

### DISCUSSION THREAD

**Discussion by vitalyjune** [apernet/hysteria#920] [Q&A] [answered]
When I start server as root it creates file in root directory: /geoip.dat
How can I set another location for it?

**Comment by haruue** [comment, marked as answer]
The geoip.dat is automatically downloaded to the [working directory](https://en.wikipedia.org/wiki/Working_directory). Change the working directory by `cd` before running the `hysteria` command to change where it was downloaded.

For an example, suppose you run hysteria like this

```
cd /etc/hysteria
/path/to/hysteria server -c config.yaml
```

This will download the geoip.dat as `/etc/hysteria/geoip.dat`.

If you are writing a systemd unit, take a look at our [AUR package](https://aur.archlinux.org/cgit/aur.git/tree/hysteria-server@.service?h=hysteria#n9) and [`man:systemd-exec(5)`](https://man7.org/linux/man-pages/man5/systemd.exec.5.html#:~:text=in%20version%20250.-,WorkingDirectory%3D,-Takes%20a%20directory).

**Reply by vitalyjune** [reply]
Thanks a lot. I use OpenRC and based on your reply found it has directive to set home directory.
For those who may need it, that is how OpenRC service script may look, it downloads geoip.dat as `/var/lib/hysteria/geoip.dat.`

```
#!/sbin/openrc-run
supervisor=supervise-daemon
name="Hysteria2"
description="powerful, lightning fast and censorship resistant proxy"
: ${config_file:="/usr/local/etc/hysteria/hysteria.yaml"}
command="/usr/local/bin/hysteria"
command_args="server -c $config_file"
command_user="hysteria"
capabilities="^cap_net_bind_service"

#that will become working directory where hysteria downloads geoip.dat 
directory=/var/lib/hysteria

depend() {
        need net
        after nftables
}
```
